libafl_tinyinst/src/shim.cc

```diff
@@ -0,0 +1,17 @@
+#include "shim.h"
+
+std::unique_ptr<LiteCov> litecov_new() {
+  return std::make_unique<LiteCov>();
+}
+
+std::unique_ptr<Coverage> coverage_new() {
+  return std::make_unique<Coverage>();
+}
+
+void get_coverage_map(uint8_t *bitmap, size_t map_size, Coverage &newcoverage) {
+  for (auto iter = newcoverage.begin(); iter != newcoverage.end(); iter++) {
+    for (auto &offset : iter->offsets) {
+      if (offset < map_size) { bitmap[offset] = 1; }
```

Comment:
It seems all offsets where offset > map_size will get ignored. I don't know how map_size is computed, but I can comment on what an `offset` is in TinyInst:

It is basically a 64-bit number uniquely representing a piece of coverage
 - In basic block coverage mode, an `offset` is a basic block offset counting from the start of the module.
 - In edge coverage mode, `offset` actually consists of 2 offsets, the offset of the previous bb in the higher 32 bits and the next bb in the lower 32 bits.
 - For cmp coverage, offset has the highest bit set and it encodes the offset of the basic block with cmp instruction, offset of that instruction within the bb and number of bytes that matched in the cmp instruction.

So in order to support more advanced forms of coverage than just basic blocks < map_size, most likely some conversion is needed between `offset` and `bitmap` encoding.